src/app/content/components/NudgeStudyTools/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { HTMLElement } from '@openstax/types/lib.dom';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { FormattedMessage } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { useAnalyticsEvent } from '../../../../helpers/analytics';
+import { useDebouncedMatchMobileQuery } from '../../../reactUtils';
+import { assertDocument, assertNotNull } from '../../../utils';
+import { closeNudgeStudyTools, openNudgeStudyTools } from '../../actions';
+import { showNudgeStudyTools } from '../../selectors';
+import { hasStudyGuides as hasStudyGuidesSelector } from '../../studyGuides/selectors';
+import arrowDesktop from './assets/arrowDesktop.svg';
+import arrowMobile from './assets/arrowMobile.svg';
+import {
+  NudgeArrow,
+  NudgeBackground,
+  NudgeCloseButton,
+  NudgeCloseIcon,
+  NudgeContent,
+  NudgeContentWrapper,
+  NudgeHeading,
+  NudgeSpotlight,
+  NudgeText,
+  NudgeWrapper,
+} from './styles';
+import {
+  setNudgeStudyToolsCookies,
+  shouldDisplayNudgeStudyTools,
+  useIncrementPageCounter,
+  usePositions,
+} from './utils';
+
+// tslint:disable-next-line: variable-name
+const NudgeStudyTools = () => {
+  const wrapperRef = React.useRef<HTMLElement>(null);
+  const body = React.useRef(assertNotNull(assertDocument().querySelector('body'), 'body element is not defined'));
```

Comment:
I'm sorry but i think it was a mistake to put it in a ref. Seems like it will query each render but just disgard the results, and you can't use a function for expensive initial value like in `useState`. Maybe it's just better to `assertDocument` at the top of component and `document.body` when needed.